### PR TITLE
feat: percentage display in partitioning charts

### PR DESCRIPTION
**Pull request by monfera**
## Summary

Adds the option of using the ratio rather than the value in `data` that's the basis for aggregation.

The ratio is computed as the value of the sector or treemap square divided by the total sum of the value.

Client code may use the ratio to display the ratio, or a percentage, instead of the sector / rectangle value.

image
image

One item in https://github.com/elastic/elastic-charts/issues/518

### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

- [x] Any consumer-facing exports were added to `src/index.ts` (and stories only import from `../src` except for test data & storybook)
- [ ] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)
- [X] Proper documentation or storybook story was added for features that require explanation or tutorials
- [ ] Unit tests were updated or added to match the most common scenarios
- [X] Each commit follows the [convention](https://github.com/elastic/elastic-charts/blob/master/CONTRIBUTING.md)

### CODE REVIEW

**Comment by markov00** [comment]
I've scanned the code and LGTM. I just want to make a small comment on the API:
your implementation impose to import a function, the `percentageValueGetter` and add them to the `valueGetter` props of the `<Partition/>` component. I think it's fine, but what if we provide a simpler version also? like allow the user to specify a string/constant or a function like

`type ValueGetter = 'value' | 'percentage' |  (node: ShapeTreeNode) => number;`

This will simplify the API a bit and will make it a bit more readable in few cases

**Comment by monfera** [comment]
image

File: src/chart_types/partition_chart/state/selectors/tooltip.ts [hunk 1]
```diff
@@ -73,7 +76,7 @@ export const getTooltipInfoSelector = createCachedSelector(
           specId: pieSpec.id,
           key: pieSpec.id,
         },
-        value: valueFormatter(valueGetterFun(shape)),
+        value: valueFormatter(primaryValueGetterFun(shape)) + ' (' + percentFormatter(percentValueGetter(shape)) + ')',
```
Comment (markov00) [inline src/chart_types/partition_chart/state/selectors/tooltip.ts]: nit: can you use a template string here?
```suggestion
        value: `${valueFormatter(primaryValueGetterFun(shape))} (${percentFormatter(percentValueGetter(shape))})`,
```

File: src/chart_types/partition_chart/layout/config/config.ts [hunk 2]
```diff
@@ -28,18 +28,18 @@ export function ratioValueGetter(node: ShapeTreeNode): number {
...
+    ? d.toExponential(Math.min(2, Math.max(0, significantDigitCount(d) - 1)))
+    : d.toLocaleString(void 0, {
+        maximumSignificantDigits: 4,
+        maximumFractionDigits: 3,
+        useGrouping: true,
+      });
+}
+
+export function percentFormatter(d: number): string {
+  return Math.round(d) + '%';
```
Comment (markov00) [inline src/chart_types/partition_chart/layout/config/config.ts]: nit: template string?
```suggestion
  return `${Math.round(d)} %`;
```

**Comment by markov00** [comment]
Hey @monfera I've pushed a change to add the `percentFormatter` as part of the spec. This because Kibana has the freedom to change the percentage format depending on the Advanced settings. you can see the changes here: https://github.com/elastic/elastic-charts/pull/558/commits/ae26259862f703f43628c789c39bcd981e5b4eff

I've also removed a small portion of code (the one related to the indices) on the tooltip selector, not sure what was the original requirements for these indices values, see: https://github.com/elastic/elastic-charts/pull/558/commits/1843a16e963ca95ae73ef13681b96abca159d7ac

**Comment by monfera** [comment]
Thanks @marco the new commits look great. We can reconsider the [`datumIndices`](https://github.com/elastic/elastic-charts/commit/6bf9a69b12d3075330a5728b7bdb4443e6244985#diff-ca54637bd0932e9f3ebba47e2b6007e9R87-R95) whenever the need comes back

**Comment by markov00** [comment]
@monfera sure, let me copy the code here also:
```
    const datumIndices = new Set();
    pickedShapes.forEach((shape) => {
      const node = shape.parent;
      const shapeNode = node.children.find(([key]) => key === shape.dataName);
      if (shapeNode) {
        const indices = shapeNode[1][INPUT_KEY] || [];
        indices.forEach((i) => datumIndices.add(i));
      }
    });
```

**Pull request merged**